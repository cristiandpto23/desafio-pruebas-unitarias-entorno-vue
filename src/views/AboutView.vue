<template>
    <div class="about">
        <h1>Acerca de nosotros</h1>
        <p>Esta es la página About de nuestra aplicación Vue.</p>
    </div>
</template>

<script>
export default {
    name: 'AboutView',
};
</script>
