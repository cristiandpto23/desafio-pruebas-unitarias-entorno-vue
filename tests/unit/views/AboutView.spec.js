import { mount } from '@vue/test-utils';
import { createRouter, createWebHistory } from 'vue-router';
import AboutView from '@/views/AboutView.vue';

describe('AboutView', () => {
    test('Probando la existencia del componente o vista AboutView', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [
                {
                    path: '/about',
                    name: 'AboutView',
                    component: AboutView,
                },
            ],
        });
        router.push('/about');
        await router.isReady();

        const wrapper = mount(AboutView, {
            global: {
                plugins: [router],
            },
        });
        expect(wrapper.findComponent(AboutView).exists()).toBe(true);
    });

    test('matches snapshot', () => {
        const wrapper = mount(AboutView);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
