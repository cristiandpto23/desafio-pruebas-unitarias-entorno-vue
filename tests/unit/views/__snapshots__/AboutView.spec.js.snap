// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView matches snapshot 1`] = `
<div class="about">
  <h1>Acerca de nosotros</h1>
  <p>Esta es la página About de nuestra aplicación Vue.</p>
</div>
`;
